packages/graphql/src/utils/execute.ts

```diff
@@ -67,7 +62,60 @@ async function execute(input: {
         }
     }
 
-    const session = input.context.driver.session(sessionParams);
+    return sessionParams;
+};
+
+const getTransactionConfig = () => {
+    const app = `${environment.NPM_PACKAGE_NAME}@${environment.NPM_PACKAGE_VERSION}`;
+    return {
+        metadata: {
+            app,
+            type: "user-transpiled",
+        },
+    };
+}
+
+const getExecutor = (input: {
+    defaultAccessMode: SessionMode;
+    context: Context;
+}): { driver?: Driver; session?: Session; transaction: Transaction } => {
+    if (!input.context.executionContext) {
+        const driver = input.context.driver;
+        const session = driver.session(getSessionParams(input));
+        const transaction = session.beginTransaction(getTransactionConfig());
+        return { driver, session, transaction };
+    }
+
+    const executionContextType = Object.getPrototypeOf(input.context.executionContext).constructor.name;
```

Comment:
See my comments below for using `instanceof`.

```suggestion
```

Reply:
Yes, I think all of your suggestions here can be used :)